src/pages/scan/ScanPage.tsx

```diff
@@ -34,12 +34,14 @@ export default function ScanPage() {
     }
     setProcessing(true);
 
-    let scannedData;
-    try {
-      scannedData = JSON.parse(decodedText);
-    } catch (e) {
-      handleError(e, 'Error parsing the QRCode data');
-      return;
+    let scannedData = await parseCustomQRCodeData(decodedText, errorModal, customQRCodes);
```

Comment:
So while my fix now un-breaks custom QR codes, the behavior is still a bit weird when you scan one matching a valid pattern, but it does not exist in Indico.

It would be nicer if in that case we showed a separate error ("Could not fetch participant data" or whatever we use if the participant does not exist), and only kept going with the usual logic of parsing the QR code as JSON etc.

My suggestion would be to do this when a custom qr code is scannerd:

- if data returned --> use that data
- if no data returned but valid json --> use that data (this helps in case someone uses a poor regex that also matches on regular ticket data)
- if no data returned and invalid json --> fail with a "no registration for custom qr code" error message

Reply:
I think making the error handling a bit nicer as suggested here is the last thing messing before this can all be merged :)